atlasdb-cassandra/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraTopologyValidatorTest.java

```diff
@@ -568,6 +568,53 @@ public void returnsAllNewHostsIfConsensusOnConfigHostsDiffersFromPreviousWhenNoQ
                 .containsExactlyInAnyOrderElementsOf(allCassandraServers);
     }
 
+    @Test
+    public void acceptsNewHostsIfConsensusOnConfigHostsIsEvolutionOfPreviousNoQuorumAcceptAndNoCurrentServers() {
+        // We split hosts into three "generations": old, new and additional. The "original hosts" refer to the old
+        // and new hosts, but not the additional ones.
+        Map<CassandraServer, CassandraClientPoolingContainer> originalHosts = initialiseHosts(ALL_HOSTS);
+        Map<CassandraServer, CassandraClientPoolingContainer> oldHosts =
+                filterServerToContainerMap(originalHosts, OLD_HOSTS::contains);
+        Set<CassandraServer> oldCassandraServers = oldHosts.keySet();
+        Set<CassandraServer> originalCassandraServers = originalHosts.keySet();
+
+        setHostIds(filterContainers(originalHosts, OLD_HOSTS::contains), HostIdResult.success(UUIDS));
+        validator.getNewHostsWithInconsistentTopologies(mapToTokenRangeOrigin(oldCassandraServers), oldHosts);
+
+        Set<String> hostsOffline = OLD_HOSTS;
+        setHostIds(filterContainers(originalHosts, hostsOffline::contains), HostIdResult.hardFailure());
+        setHostIds(
+                filterContainers(originalHosts, server -> !hostsOffline.contains(server)),
+                HostIdResult.success(MAYBE_SAME_AS_FIRST_CLUSTER));
+
+        configServers.set(NEW_HOSTS);
+        assertThat(validator.getNewHostsWithInconsistentTopologies(
+                        splitHostOriginBetweenLastKnownAndConfig(
```

Comment:
I could be misreading this, but: In L582 we've validated that all the oldCassandraHosts were valid, so in normal operations, we wouldn't then add the oldCassandraServers in the _new_ servers list. (and same for the last assertion)

I flag this because it did seem a bit weird that we're returning inconsistent topology nodes in a test about accepting new nodes (at least, personally I'd find it a bit easier to read if the assertions verified there were no new nodes with inconsistent topologies, unless you were explicitly testing a case where some of the new nodes had a different (and inconsistent topology), but that didn't seem to be the case)

Reply:
Yep - "noCurrentServers" isn't really a situation that can exist now outside of startup, since we changed the handling of absentServers. I've changed the test to be one with no quorum but with the previous set of current servers at each stage.